src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -682,27 +682,19 @@ public void setQuorumAddress(InetSocketAddress addr){
     }
 
     public InetSocketAddress getElectionAddress(){
-        synchronized (QV_LOCK) {
-            return myElectionAddr;
-        }
+        return myElectionAddr;
```

Comment:
Ideally we'd want to pass precise information to another peer but under current design it sounds not practical to me to break the order of acquiring locks. A redesign of lock free / reactive style communication between peer and qcm might fix the root cause this.